defi/src/utils/shared/distressedCoins.ts

```diff
@@ -0,0 +1,136 @@
+import { chainsThatShouldNotBeLowerCased } from "../../utils/shared/constants";
+import { elastic } from "@defillama/sdk";
+import { getClient as getCoinsESClient } from "../../storeTvlInterval/computeTVL";
+import { runInPromisePool } from "@defillama/sdk/build/generalUtil";
+import { batchWrite } from "./dynamodb";
+
+const client: any = elastic.getClient(); 
+
+function sanitizeKey(key: string): string {
+  const chain = key.split(":")[0];
+  const address = key.substring(chain.length + 1);
+  const normalizedAddress = chainsThatShouldNotBeLowerCased.includes(chain) ? address : address.toLowerCase();
+  return `${chain}:${normalizedAddress}`;
+}
+
+// batch check if a list of coins are distressed
+export async function batchIsDistressed(keys: string[]) {
```

Comment:
can you add the isDistressed field to the price api response, I will update the sdk to include that in getPrices response

then, for the batch call here, we check if the metadata response has `isDistressed` set to true